src/protocol.utility.ts

```diff
@@ -31,4 +33,39 @@ export class ProtocolUtility {
 
         return result;
     }
+
+    /*
+        Consumers can call this method to retry logic until a duration has passed or the retry logic
+        returns true.
+
+        durationMS:  how long the retry will occur before it throws an exception
+        waitBetweenMD: delay between iterations
+        retryFunction: async method containing retry logic.  On true, the retry loop exists.  On false it retries
+
+         toThink(): should retryFunction return 3 possible values:
+            -1 abort logic and throw exception (no current value for this)
+             0 to continue retry (currently false)
+             1 to exit retry and return true (currently true)
+     */
+    public static async retryForDuration(durationMS: number, waitBetweenMS: number, retryFunction: any): Promise<boolean> {
+        if (!retryFunction) {
+            throw new ProtocolException('Common Error', 'retryFunction is required');
+        }
+
+        const startOf = new Date();
+        while (durationMS > ProtocolUtility.timeDelta(new Date(), startOf)) {
+            try {
+                if (true === await retryFunction()) {
```

Comment:
will retryFunction always return true? I think in some cases it will return the results of an api call

Reply:
The purpose of the retryFunction is to a) do whatever you want to retry and b) inform the loop logic if it should continue or not.  Returning the results of the actual retry code (aka return of API call) is meaningless to the loop logic, it will not know how to process that information.   

I think if the consumer (aka implementation of retryFunction) needs to bubble up information back to its own implementation, it will have to find another way to communicate that.